### ISSUE TITLE
Publicize: Add Instagram as a service

### ISSUE THREAD

**Issue by richardmtl** [Automattic/jetpack#1870]
As requested in 2137808-t

**Comment by jkudish** [comment]
This isn't possible. Instagram doesn't allow publishing to them via any kind of API.

**Comment by faddah** [comment]
@richardmtl @jkudish @jeherve —

could this please be re-opened and reviewed for future JetPack enhancement. the API policy at Instagram has changed, so that now, like Twitter, if you authorize the app/service through Instagram OAuth, you may, indeed, post to Instagram from another app/service. See [this article here](http://stackoverflow.com/questions/18242610/uploading-photos-to-instagram-via-your-own-ios-app). Please re-open & consider? Thank you.

**Comment by kraftbj** [comment]
Do you have a link to any policy change? That post's answer states `Instagram doesn't have a public API for posting photos, it lets you get their data but not write your own.`

Looking at the documentation at https://instagram.com/developer/ , the closest is iOS/Android apps can push photos to the Instagram app for that app to push up to Instagram.

**Comment by jkudish** [comment]
> Looking at the documentation at https://instagram.com/developer/ , the closest is iOS/Android apps can push photos to the Instagram app for that app to push up to Instagram.

That's correct. They still don't (and probably never will) allow direct publishing from their API.

**Comment by RavanH** [comment]
https://wordpress.org/plugins/wp-posts-to-instagram-by-kolesyane/

**Comment by jeherve** [comment]
@RavanH Thanks! It looks like this plugin works around the problem by sending the requests and authenticating itself as an Android device. While that may work for a while, I'm not sure it's really in line with Instagram's platform policy.

**Comment by RavanH** [comment]
Ah yes indeed, now that you mention it, I see API_AGENT in there too...

**Comment by dougaitken** [comment]
Adding here after someone asked today. This is technically possible now with the [Instagram Graph API](https://developers.facebook.com/blog/post/2018/01/30/instagram-graph-api-updates/)

> Content Publishing Beta [NEW] - Available to Facebook Marketing Partners and Instagram Partners only. Post an image to Feed with or without a caption.

(Instagram biz profiles only so very similar to current situation with publicize and Facebook )

**Comment by kraftbj** [comment]
I'm tempted to close it for now. Based on https://developers.facebook.com/docs/instagram-api/content-publishing , it is in closed beta and not taking on new partners right now.

Anything we build right now would be subject to breakage if they change the API and we likely want to wait until it is stable.

**Comment by dougaitken** [comment]
I didn't read into that. Good point, @kraftbj. Closing for now until the API is opened up

**Comment by Josimarcsilva** [comment]
How can hootsuite and social zoho manage to make posts for Instagram? Would not it be the case Jetpack do the same thing?

Instagram has an official API
https://developers.facebook.com/docs/instagram/sharing-to-feed

What I do not understand?

**Comment by jeherve** [comment]
@Josimarcsilva We've covered those points in the comments above. That should clarify things a bit!

- The Instagram API is open to mobile apps only.
- Instagram also offers an Instagram Graph API, but that API is only available when accessing data in large- and medium-sized Instagram Business Accounts.

**Comment by jeherve** [comment]
Also in 2428533-zen

**Comment by sitestudioapp** [comment]
Buffer, MailChimp, SocialBee, CoSchedule (A WordPress plugin), Hootsuite, Later, Sprout Social, Friends+Me and basically every other social media positing tool have this feature (Web and mobile), some for 2 years now. Odd that this is such a challenge for Jetpack.

**Comment by indolering** [comment]
> @sitestudioapp Odd that this is such a challenge for Jetpack.

Hootsuite appears to be able to [publish from the dashboard](https://hootsuite.com/instagram) but Buffer still uses a [reminder work-around](https://buffer.com/instagram).  I'm assuming Hootsuite was grandfathered in.

> @jeherv The Instagram API is open to mobile apps only.

You do have a mobile app and could implement the standard reminder workaround, which would also be helpful with Snapchat.  Let's move that discussion to a new ticket: [Share Reminders](https://github.com/Automattic/jetpack/issues/15326) 😁.

> @jeherve Instagram also offers an Instagram Graph API, but that API is only available when accessing data in large- and medium-sized Instagram Business Accounts.

[The documentation states](https://developers.facebook.com/docs/instagram-api) that you only need a professional account, which isn't even limited to businesses.  I have one and it's just a matter of [changing your account settings](https://help.instagram.com/502981923235522).

>  @kraftbj Anything we build right now would be subject to breakage if they change the API and we likely want to wait until it is stable.

It's a pretty simple REST API that [hasn't changed in two years](https://web.archive.org/web/20190701225605/https://developers.facebook.com/docs/instagram-api/guides/content-publishing).

> @kraftbj I'm tempted to close it for now. Based on https://developers.facebook.com/docs/instagram-api/content-publishing, it is in closed beta and not taking on new partners right now.

That stance is two years old and Wordpress is kind of a big deal.  Facebook rejecting an OSS project that's central to the web ... that's gotta be bad press for a company under anti-trust investigations.   Maybe ask Matt to throw some weight around?

**Comment by kraftbj** [comment]
Similar to Facebook's stance on removing access to post on personal timelines, we don't expect out clout to influence much here.

**Comment by indolering** [comment]
FWIW, Meta has [opened up content publishing to Instagram](https://developers.facebook.com/docs/instagram-api/guides/content-publishing/?translation).  The only catch is that the account being posted to must be a business account.

**Comment by kalimba666** [comment]
The wordpress plugin blog2social does it, too.

And as @indolering said, it is open for business accounts (and it's very easy to transform an account into a business acc). So implementing should not be a problem anymore.
